client/src/utils/parser/robotCodeToSsotParsing/robotCodeToSsotParsing.js

```diff
@@ -0,0 +1,411 @@
+ * @description deletes everything before the first occurence of '=' and then trims all emptyspace until the rpa task name to get the expected format
+ * @param {String} currentLine current line of RPAf code
+ * @param {String} splitPlaceholder placeholder to split the string
+ * @returns the current line without the outputVariableName prefix as string
+ */
+const currentLineWithoutOutputVariableName = (completeLine, splitPlaceholder) => {
+    const indexOfEqualsSign = completeLine.indexOf('=');
+    let currentLine = completeLine.slice(indexOfEqualsSign + 1);
+    if (currentLine.startsWith(splitPlaceholder)) {
+        currentLine = currentLine.replace(splitPlaceholder, '').trim();
+    } else {
+        currentLine = currentLine.trim();
+    }
+    return currentLine;
+}
+
+/**
+ * @description "preprocesses" the code in a usable data format
+ * @param {Array} robotCodeTaskSection robot code w/o empty lines as an array of Strings
+ * @param {Array} declaredApplications all declared Aplications from ***settings*** section as Strings 
+ * @returns Array of Objects with the following schema:
+ *      instructionBlocks = [rpaApplication:String, rpaTask:String, name:String, paramArray:Array]
+ */
+const getInstructionBlocksFromTaskSection = (robotCodeTaskSection, declaredApplications) => {
+    let currentApplication;
+    let errorWasThrown;
+    const instructionBlocks = [];
+    const regexForElementNameLine = (/#/)
+    const regexForOutputVariable = (/\${(.)+} =/)
+    const splitPlaceholder = '§&§';
+
+    robotCodeTaskSection.slice(1).forEach((line) => {
+        if (errorWasThrown) return;
+        let currentLine = line.trim();
+        const currentLineDefinesNewApplication = declaredApplications.includes(currentLine);
+        const currentLineContainsElementName = regexForElementNameLine.test(currentLine);
+        const currentLineIncludesSplitPlaceholder = currentLine.includes(splitPlaceholder);
+        const currentLineHasNoSpecifiedApplication = typeof currentApplication === 'undefined';
+        const currentLineDefinesOutputValue = regexForOutputVariable.test(currentLine);
+
+        if (currentLineDefinesNewApplication) {
+            currentApplication = currentLine;
+            return;
+
+        } if (currentLineHasNoSpecifiedApplication) {
+            customNotification('Error', `There is no RPA-Application specified for line "${currentLine}"`)
+            errorWasThrown = true;
+            return;
+        }
+
+        if (currentLineIncludesSplitPlaceholder) {
+            customNotification('Error', `It is not allowed to use & or § as param values \nError location: "${line}"`)
+            errorWasThrown = true;
+            return;
+        }
+
+        if (currentLineContainsElementName) {
+            instructionBlocks.push({ rpaApplication: currentApplication, name: currentLine.substring(1).trim() })
+            return;
+        }
+
+        currentLine = currentLine.replaceAll(FOURSPACE, splitPlaceholder);
```

Comment:
Please never use replaceAll. (Btw, that's one reason why the tests are failing in the pipeline)
Please also check for other occurrences with the search function, I will not comment on every replaceAll